runtime/jcl/common/sun_misc_Unsafe.cpp

```diff
@@ -788,6 +788,58 @@ Java_jdk_internal_misc_Unsafe_objectFieldOffset1(JNIEnv *env, jobject receiver,
 	return offset;
 }
 
+void JNICALL
+Java_jdk_internal_misc_Unsafe_writebackMemory(JNIEnv *env, jobject receiver, jlong addr, jlong len)
+{
+	J9VMThread *currentThread = (J9VMThread*)env;
+	J9JavaVM *vm = currentThread->javaVM;
+	J9InternalVMFunctions *vmFuncs = vm->internalVMFunctions;
+
+#if defined(LINUX) && (defined(J9X86) || defined(J9HAMMER))
+	PORT_ACCESS_FROM_ENV(env);
+	J9ProcessorDesc desc;
+	j9sysinfo_get_processor_description(&desc);
+	uintptr_t cache_line_size = desc.features[2] & 0xFF00; /* bits 8-15 contain cache line size */
+	U_8 *end_addr = (*(U_8 **) &addr) + (*(uintptr_t *) &len);
+	U_8 *ptr = *(U_8 **)&addr;
+	if (j9sysinfo_processor_has_feature(&desc, J9PORT_X86_FEATURE_CLWB)) {
+		issueReadWriteBarrier();
+		for (; ptr < end_addr; ptr += cache_line_size) {
+			asm volatile("clwb %0" : "+m" (*ptr));
+		}
+		issueReadWriteBarrier();
+		return;
+	} else if (j9sysinfo_processor_has_feature(&desc, J9PORT_X86_FEATURE_CLFSH)) {
+		issueReadWriteBarrier();
+		for (; ptr < end_addr; ptr += cache_line_size) {
+			asm volatile("clflush %0" : "+m" (*ptr));
+		}
+		issueReadWriteBarrier();
+		return;
+	}
+#endif /* linux and x86 */
+
+	vmFuncs->internalEnterVMFromJNI(currentThread);
+	vmFuncs->setCurrentException(currentThread, J9VMCONSTANTPOOL_JAVALANGRUNTIMEEXCEPTION, NULL);
+	vmFuncs->internalExitVMToJNI(currentThread);
+}
+
+jboolean JNICALL
+Java_jdk_internal_misc_Unsafe_isWritebackEnabled(JNIEnv *env, jobject receiver)
+{
+	jboolean ret = JNI_FALSE;
+#if defined(LINUX) && (defined(J9X86) || defined(J9HAMMER))
+	PORT_ACCESS_FROM_ENV(env);
+	J9ProcessorDesc desc;
+	j9sysinfo_get_processor_description(&desc);
+	if (j9sysinfo_processor_has_feature(&desc, J9PORT_X86_FEATURE_CLWB) ||
+		j9sysinfo_processor_has_feature(&desc, J9PORT_X86_FEATURE_CLFSH)) {
+		ret = JNI_TRUE;
```

Comment:
Does this require OS support as well?  The JEP mentions MAP_SYNC which implies a particular OS level is required

Reply:
No, there is no OS support needed.